packages/core/src/Site/index.js

```diff
@@ -844,9 +851,39 @@ class Site {
   }
 
   async reloadSiteConfig() {
+    const oldSiteConfig = this.siteConfig;
     const oldAddressablePages = this.addressablePages.slice();
     const oldPagesSrc = oldAddressablePages.map(page => page.src);
     await this.readSiteConfig();
+    await this.handleIgnoreReload(oldSiteConfig);
+    await this.handlePageReload(oldAddressablePages, oldPagesSrc,
+                                oldSiteConfig, this.isGlobalConfigModified(oldSiteConfig));
+    await this.handleStyleReload(oldSiteConfig.style);
+  }
+
+  /**
+   * Checks if any of the global attributes of site.json is modified
```

Comment:
```suggestion
   * Checks if any attributes of site.json requiring a global rebuild are modified
```
`global attributes` might be a little vague